lib/MySQL_FTS.cpp

```
@@ -0,0 +1,768 @@
+
+MySQL_FTS::~MySQL_FTS() {
+	close();
+}
+
+int MySQL_FTS::init() {
+	// Initialize database connection
+	db = new SQLite3DB();
+	char path_buf[db_path.size() + 1];
+	strcpy(path_buf, db_path.c_str());
+	int rc = db->open(path_buf, SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE);
+	if (rc != SQLITE_OK) {
```

Cppcheck (2.19.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)